ext/pgsql/pgsql.c

```diff
@@ -973,23 +972,22 @@ PHP_FUNCTION(pg_ping)
 PHP_FUNCTION(pg_query)
 {
 	zval *pgsql_link = NULL;
-	char *query;
+	zend_string *query;
```

Comment:
Most of the zend_string changes don't make sense to me. Using zend_string makes sense if you actually use it as such, e.g. pass it around as zend_string, copy it, etc. Here you are just accepting the zend_string and then always using ZSTR_VAL, which imho is worse than the previous code.

Reply:
I was going on a top-down approach to start pass zend_strings to more internal functions, will continue to work on that and revisit the cases where that doesn't occur then.

Reply:
Right. It's mostly about cases like this, where a string is only used to talk to some 3rd-party C API (libpq), which is never going to accept zend_strings, so it just adds extra friction.